tendrl/node_agent/node_sync/sds_detect.py

```diff
@@ -94,42 +95,50 @@ def sync(sync_ttl):
                                 _cluster.status = "new_peers_detected"
                                 _cluster.save()
-                            integration_id = NS.tendrl_context.integration_id
-                            nodes_ids = json.loads(etcd_utils.read(
-                                "indexes/tags/tendrl/integration/%s" %
-                                integration_id
-                            ).value)
-                            cluster_peers = []
-                            try:
-                                cluster_peers = etcd_utils.read(
-                                    "/clusters/%s/Peers" % integration_id
-                                )
-                            except etcd.EtcdKeyNotFound:
-                                pass
-                            peer_count = 0
-                            for peer in cluster_peers.leaves:
-                                peer_count += 1
                             _cluster = NS.tendrl.objects.Cluster(
-                                integration_id=integration_id
+                                integration_id=NS.tendrl_context.integration_id
                             ).load()
-                            if len(nodes_ids) == peer_count and \
-                                _cluster.status == "new_peers_detected":
-                                # All the nodes are having node-agents running
-                                # and known to tendrl
-                                msg = "New nodes in cluster: %s have node " \
-                                    "agents running now. Cluster is ready " \
-                                    "to expand." % integration_id
-                                event_utils.emit_event(
-                                    "cluster_status",
-                                    "expand_pending",
-                                    msg,
-                                    "cluster_{0}".format(integration_id),
-                                    "INFO",
-                                    integration_id=integration_id
+                            if _cluster.status == "new_peers_detected":
+                                peers = []
+                                cmd = subprocess.Popen(
+                                    "gluster pool list",
+                                    shell=True,
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE
                                 )
-                                # Set the cluster status accordingly
-                                _cluster.status = 'expand_pending'
-                                _cluster.save()
+                                out, err = cmd.communicate()
+                                if err or out is None or \
+                                    "Connection failed" in out:
+                                    pass  # set the no of peers as zero
```

Comment:
I think you wanted to use "continue" or "return"?

Reply:
Actually its fine if we pass as in that case `peers` would be empty list and the condition at L#121 will not pass and no clearing alert raised. We dont want to stop the logic from L#143 from execution in this case. Correct?

Reply:
L#113 will raise exception if out is None, handle that and then the pass makes sense

Reply:
Ah yes. will add a check